cg/apps/gisaid/gisaid.py

```diff
@@ -0,0 +1,126 @@
+"""Interactions with the gisaid cli upload"""
+
+import logging
+from pathlib import Path
+from typing import List
+
+from cg.store import models
+from cg.meta.workflow.fastq import FastqHandler
+from ...utils import Process
+from .models import GisaidSample, UpploadFiles
+from housekeeper.store import models as hk_models
+
+from cg.meta.meta import MetaAPI
+
+LOG = logging.getLogger(__name__)
+
+import csv
+
+
+class GisaidAPI(MetaAPI):
+    """Interface with Gisaid cli uppload"""
+
+    def __init__(self, config: dict):
+        super().__init__(config)
+        self.gisaid_submitter = config["gisaid"]["submitter"]
+        self.gisaid_binary = config["gisaid"]["binary_path"]
+        self.process = Process(binary=self.gisaid_binary)
+        self.fastq_handler = FastqHandler()
+
+    def get_headers(self) -> list:
+        """Get GisaidSample keys as list."""
+
+        return list(GisaidSample.schema().get("properties").keys())
+
+    def get_sample_row(self, sample: models.Sample, family_id: str, fasta_header: str) -> List[str]:
+        """Build row for a sample in the batch upload csv."""
+
+        orig_lab: str = "Stockholm"  # sample.originating_lab
+        collection_date: str = "201122"  # sample.collection_date str or date, we dont know yet
```

Comment:
Let's try to use ISO formated dates as much as possible. That will make it easier for every library. (YYYY-MM-DD)

Reply:
is type str then or datetime.date.isorformat typ?